pkg/kubelet/apis/kubeletconfig/validation/validation.go

```diff
@@ -86,6 +87,19 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if kc.RegistryPullQPS < 0 {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: RegistryPullQPS (--registry-qps) %v must not be a negative number", kc.RegistryPullQPS))
 	}
+	// since EvictionHard has a non-empty default, we provide a "none" option that can be used to disable it
+	// when this option is specified, no other thresholds should be present
+	for signal, _ := range kc.EvictionHard {
+		if len(kc.EvictionHard) > 1 && evictionapi.Signal(signal) == evictionapi.SignalNone {
+			allErrors = append(allErrors, fmt.Errorf("invalid configuration: EvictionHard (--eviction-hard) %v may not contain additional thresholds when '%s' is specified", evictionapi.SignalNone))
+		}
+	}
+	// we do the same for EvictionSoft, for consistent behavior
```

Comment:
I think we should either have `eviction-soft-grace-period` accept a `none` value as well, or skip soft eviction for now.  This is because you need to specify a soft eviction grace period for each soft eviction threshold: https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/#soft-eviction-thresholds

Reply:
If you set 'none' in EvictionSoft, softThresholds will be nil, and we won't require a corresponding grace period. It doesn't make sense to me to require 'none' to be specified in two places, when the second place's semantics depend on the first's.